[Ong Jingwen] iP

Jingwencloud opened the pull request:
## Duke
> “_What we fear doing most is usually what we most need to do._” – Tim Ferriss (source)

Duke keeps track of what you need to do and frees your mind of having to remember them. It's,

- text-based
- easy to learn
- **FAST SUPER FAST** to use

All you need to do is,

1. download it from [here.](https://github.com/Jingwencloud/ip/releases/tag/v0.1)
2. Open a command window in that folder.
3. Run the command `java -jar {filename}.jar`
4. Begin using Duke 😃 

let it manage your tasks for you 😉
And it is FREE! :dancer:

Features:

- [x] Managing tasks
- [x] Managing deadlines
- [x] Managing events

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

    public static void main(String[] args) {
        new Duke("data/tasks.txt").run();
    }

Commit:
Add Gradle support

Jingwencloud committed:
Update Duke class to greet, echo and exit

Jingwencloud committed:
Update Duke to add and list text

Jingwencloud committed:
Add Task class

Jingwencloud committed:
Update Duke to be able to mark tasks as done

Jingwencloud committed:
Add deadline, event and todo class and update Duke to support multiple task types

Jingwencloud committed:
Add testing of Duke using I/O redirection technique

Jingwencloud committed:
Remove displaying of the string "todo" unnecessarily

Jingwencloud committed:
Update expected.txt for testing

Jingwencloud committed:
Update Duke to handle errors and update EXPECTED.TXT outputs

The task type ( todo, deadline and event) was previously included when a task description is printed. This is edited and Duke no longer prints out the task type in full.

Jingwencloud committed:
Update Duke to handle more kinds of errors

Throw DukeException when the following occur:
1. Description of any task type is empty, including case of when corresponding deadline/period is stated with no task description
2. When a word which Duke does not understand is inputted (does not start with todo, deadline, event and is not list or bye)
3. When user attempts to mark a task done when the number inputted is invalid (eg. 4 tasks but user inputs done 5)
4. Deadline/period for event is not indicated correctly. (/by and /at not used respectively, or when no deadline/period is stated for the corresponding task)

Jingwencloud committed:
Update Duke to be able to delete tasks

Duke can delete tasks and an exception will be thrown when the task number in input is not valid, or when the delete word is not followed by a task number.

Jingwencloud committed:
Edit code to be more succinct, edit behavior of Duke and update EXPECTED.TXT

Jingwencloud committed:
Update Duke class to use Enum

Jingwencloud committed:
Add save functionality for Duke

Jingwencloud committed:
Update Duke, Deadline and Event classes to use date and time

Jingwencloud committed:
Merge branch 'branch-Level-8'

Jingwencloud committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/Duke.java

Jingwencloud committed:
Add Ui class to deal with interactions with user

Jingwencloud committed:
Add Storage class to load and save file content

Jingwencloud committed:
Add Ui, Parser and TaskList classes

Jingwencloud committed:
Update code to use more OOP

Jingwencloud committed:
Update Duke to use the new classes created

Jingwencloud committed:
Add Duke package

Jingwencloud committed:
Add JUnit tests

Jingwencloud committed:
Update printing of divider

Jingwencloud committed:
Add javadocs

Jingwencloud committed:
Edit code to comply with coding standard

Jingwencloud committed:
Edit code to follow coding standard

Jingwencloud committed:
Add find function for Duke to find matching tasks

Jingwencloud committed:
Merge branch 'branch-A-JavaDoc'

Jingwencloud committed:
Merge branch 'branch-A-CodingStandard'

Jingwencloud committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/Deadline.java
#	src/main/java/duke/Parser.java
#	src/main/java/duke/Storage.java

Jingwencloud committed:
Change format which date is stored in inside Task file

Jingwencloud committed:
Update delete function

Jingwencloud committed:
Add more javadocs

Jingwencloud committed:
Remove Tasks.txt

Jingwencloud committed:
Add gitignore for Tasks.txt file

Jingwencloud committed:
Organize into different packages

Jingwencloud committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-A-Gradle

Jingwencloud committed:
Merge branch 'branch-A-Gradle'

Jingwencloud committed:
Update build.gradle

wangpalex commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,181 @@
+package duke;
+
+import duke.command.*;
```
Importing classes under `duke.command`  one by one will make it easier to identify and improves readability 😄

Superbestron replied:
Yes I believe prof has cautioned against the use of wildcard imports.

wangpalex commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,181 @@
+
+        case "todo":
+        case "event":
+        case "deadline":
```
You can try to mark `// fall through` in these 2 lines to make it notable.

Superbestron replied:
Agreed.

wangpalex commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,181 @@
+            String[] splitTask2 = splitInput2[1].split(" +");
+            String[] splitTask = taskType.equals("deadline")
+                    ? splitInput2[1].split("/by") :
+                    splitInput2[1].split("/at");
```
Putting the `:` to next line will make it nicer and conforms coding standard.

wangpalex commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,186 @@
+     * @return The string representing the task.
+     */
+    public String fileString(Task task) {
+        String toAdd = task.taskIndicator() + " | " +
```
Break line before operators (`+`) would be more in accordance to coding standard.

wangpalex commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,73 @@
+     */
+    public void delete(int num) {
+
+        taskList.remove(num-1);
```
It is better to leave some spaces between operators and characters i to improve readability.

wangpalex commented:
LGTM. Good job! 😃

Jingwencloud committed:
Add checkstyle files and use checkstyle to detect coding violations

Superbestron commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,280 @@
+public class Ui {
+
+    /** The indentation for 4 spaces */
+    public final static String INDENT_1 = "    ";
```
I think better names could have been given for `INDENT_1`, same with `INDENT_2`

Superbestron commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,72 @@
+     * @return The string representing the date.
+     */
+    public String changeDateFormat() {
+
```
I'm not sure if this extra blank line here is intended? Same with the other few extra lines below.

Superbestron commented:
LGTM as well! Keep up the good work 👍

Jingwencloud committed:
Use checkstyle and update code to follow style rules

Jingwencloud committed:
Merge branch 'branch-A-CheckStyle'

Jingwencloud committed:
Add GUI to Duke

Jingwencloud committed:
Remove unused parts of code

Jingwencloud committed:
Merge branch 'branch-Level-10'

Jingwencloud committed:
Update GUI to greet user and add colour for GUI

Jingwencloud committed:
Update code based on PR reviews and edit Duke's dialog with user

Jingwencloud committed:
Remove unused parts of code

Jingwencloud committed:
Use assertions

Assertions are not used in the code, even though there are assumptions
about the program state which can be defined.

Assertions are used to define assumptions about the program state so
that the runtime can verify them. It can help in identifying possible
bugs in the code.

Add assertions in the relevant command classes and parser class to
define the assumptions of the commands.

Jingwencloud committed:
Improve code quality

The code quality has some areas for improvement, where there exists
deep nesting and complicated expressions in the code, as well as other
traits which reduce readability.

Improving code quality can increase readability.

Use refactoring and other techniques to improve code quality.

Jingwencloud committed:
Merge pull request #2 from Jingwencloud/branch-A-CodeQuality

Improve code quality

Jingwencloud committed:
Merge branch 'master' of https://github.com/Jingwencloud/ip

Jingwencloud committed:
Merge branch 'master' into branch-A-Assertions

# Conflicts:
#	src/main/java/duke/Parser.java
#	src/main/java/duke/command/AddCommand.java
#	src/main/java/duke/command/DeleteCommand.java

Jingwencloud committed:
Merge pull request #1 from Jingwencloud/branch-A-Assertions

Use assertions

Jingwencloud committed:
Merge branch 'master' of https://github.com/Jingwencloud/ip

Jingwencloud committed:
Set up Continuous Integration

Jingwencloud committed:
Add ability to display tasks in the form of a schedule

The tasks can only be viewed together in the form of a list.

Adding the schedule command allows users to view the tasks they have
for a specific date.

Add scheduleCommand class to allow users to view their schedule.

Jingwencloud committed:
Merge branch 'B-ViewSchedules'

Jingwencloud committed:
Refactor code for Storage class

Jingwencloud committed:
Update and add javadocs

Jingwencloud committed:
Change method names to verb form

Jingwencloud committed:
Fix bug in deleting task

Jingwencloud committed:
Add javadoc for new methods in Parser class

Jingwencloud committed:
Edit GUI for duke to be more pleasant-looking

Jingwencloud committed:
Update Duke responses

Jingwencloud committed:
Update duke response for delete command

Shows the number of tasks after a new line.

Jingwencloud committed:
Edit README.md for duke

Jingwencloud committed:
Update Ui.png image

Jingwencloud committed:
Update Ui.png

Jingwencloud committed:
Update application to exit seconds after bye message is displayed

Jingwencloud committed:
Fix checkstyle errors

Jingwencloud committed:
Update dialog box to prevent cutting off

Jingwencloud committed:
Update GUI

Jingwencloud committed:
Update Ui.png to new GUI image

Jingwencloud committed:
Fix no description error message bug

Jingwencloud committed:
Add spacing between words for done and delete command error messages
